0-bootstrap/modules/jenkins/main.tf

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cicd_project_name       = format("%s-%s", var.project_prefix, "cicd")
+  jenkins_apis             = ["cloudkms.googleapis.com"]
+  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
+  activate_apis               = distinct(var.activate_apis)
+  jenkins_gce_fw_tags         = ["ssh-jenkins-agent"]
+}
+
+/******************************************
+  CICD project
+*******************************************/
+module "cicd_project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 7.0"
```

Comment:
nit: can we use latest current versions just so that we are upto date

Reply:
I could fix this in a later version, once I have all the parts working, so I don't break dependencies. Adding a TODO here